client/ts/src/client.ts

```diff
@@ -478,6 +555,67 @@ export class ManifestClient {
       },
     );
   }
+
+  /**
+   * CancelAll instruction. Cancels all orders on a market
+   *
+   * @returns TransactionInstruction
+   */
+  public cancelAllIx(): TransactionInstruction {
+    return createBatchUpdateInstruction(
+      {
+        payer: this.payer,
+        market: this.market.address,
+        manifestProgram: MANIFEST_PROGRAM_ID,
+        owner: this.payer,
+        wrapperState: this.wrapper.address,
+      },
+      {
+        params: {
+          cancels: [],
+          cancelAll: true,
+          orders: [],
+          traderIndexHint: null,
+        },
+      },
+    );
+  }
+
+  /**
+   * ClearOutMarketTxs transactions. Pulls all orders
+   * and withdraws all balances from the market in two transactions
+   *
+   * @param payer PublicKey of the trader
+   *
+   * @returns TransactionSignatures[]
+   */
+  public async clearOutMarketTxs(
+    payerKeypair: Keypair,
+  ): Promise<TransactionSignature[]> {
+    await this.market.reload(this.connection);
+    const cancelAllIx = this.cancelAllIx();
+    const cancelAllTx = new Transaction();
+    const cancelAllSig = await sendAndConfirmTransaction(
+      this.connection,
+      cancelAllTx.add(cancelAllIx),
+      [payerKeypair],
+      {
+        commitment: 'confirmed',
+      },
+    );
+    await this.market.reload(this.connection);
```

Comment:
Leave a TODO to merge this into one TX. Can either do client side or have a feature request for the wrapper to expose this behavior.